app/src/main/java/org/mozilla/fenix/perf/ProfilerDialogFragment.kt

```diff
@@ -0,0 +1,455 @@
+            profiler = context.components.core.engine.profiler!!
+            if(!profiler.isProfilerActive()){
+                StartProfileDialog(profiler){
+                        filters, features ->
+                    context.components.core.engine.profiler!!.startProfiler(filters,features)
+                }
+            } else {
+                val profilerErrorDialog = remember { mutableStateOf(false) }
+                var errorMessage: String? = null
+                StopProfilerDialog{
+                    saveType ->
+                    context.components.core.engine.profiler!!.stopProfiler(
+                        onSuccess = {
+                            val profileByteArray = it!!
+                            if(saveType == resources.getString(R.string.profiler_save_method_url)){
+                                val url = saveProfileAsURL(profileByteArray, context)
+                                val clipboardManager = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+                                val clipData = ClipData.newPlainText("Profiler URL", url)
+                                clipboardManager.setPrimaryClip(clipData)
+                                Toast.makeText(context, "URL saved to clipboard successfully $url", Toast.LENGTH_LONG).show()
+                            }else{
+                                val intent = Intent(Intent.ACTION_CREATE_DOCUMENT).apply {
+                                    addCategory(Intent.CATEGORY_OPENABLE)
+                                    type = "application/text"
+                                    putExtra(Intent.EXTRA_TITLE, "profiler.json")
+
+                                    // Optionally, specify a URI for the directory that should be opened in
+                                    // the system file picker before your app creates the document.
+                                    putExtra(DocumentsContract.EXTRA_INITIAL_URI, "profiler.json")
+                                }
+                                profileSaveLauncher.launch(intent)
+                            }
+                        },
+                        onError = { error ->
+                            error.message?.let{
+                                errorMessage = it
+                            } ?.let {
+                                errorMessage = ""
+                            }
+                            profilerErrorDialog.value = true
+                        }
+                    )
+                    this@ProfilerDialogFragment.dismiss()
+                }
+                if(profilerErrorDialog.value){
+                    ProfilerErrorDialog(errorMessage!!)
+                }
+            }
+        }
+        }
+    }
+
+    override fun dismiss() {
+        super.dismiss()
+        findNavController().previousBackStackEntry?.savedStateHandle?.set("profilerActive",profiler.isProfilerActive())
+
+
+    }
+    @Composable
```

Comment:
issue: you raised a concern that this PR felt bloated. I think this file might feel bloated because this file has too many purposes: by definition from the Android framework, a Fragment's responsibility is to receive lifecycle events and hook that up to other components (for example, inflating the view, passing it state, and forwarding user input events to the model and back to the views). However, this file and class is including a second responsibliity: defining views.

suggestion: I think if you moved these `@Composable` lines into a separate file, this file would feel much more manageable (e.g. it goes from ~455 lines to ~131 lines).